File: Practice/atcoder/ABC/028/src/c2.rb
# 全探索する
arr = gets.chomp.split.map(&:to_i)
results = []
arr.size.times do |i|
  for j in (i+1...arr.size) do
    for k in (j+1...arr.size) do
      results << arr[i] + arr[j] + arr[k]
    end
  end
end
results = results.uniq.sort
puts results[-3]

File: Practice/atcoder/ABC/100/src/c2.py
# 考察微塵もしてないけど2の約数の数っぽい

n = int(input())
arr = list(map(int, input().split()))
count = 0
for a in arr:
  while a % 2 == 0 and a > 0:
    count += 1
    a //= 2
print(count)